ekundayo-ab/hello-books
# #153553440 Implement First Code Review Feedback

### What does this PR do?
Ensures code review feedback are implemented

### Description of Task to be completed?
- setup webpack to make application ready for production by uglifying code, extract css file and make webpack config a single file loading configurations based on application environment
- ensure tests are comprehensive enough, improve e2e-test by adding more assertions, ensure some test descriptions match what is tested and increase number of tests on client side for more code coverage
- remove character sequences from api documentation and improve on application ReadMe
- ensure pagination component does not flicker when used and disallow improper white space margin on pages with small content
- ensure application does not return `book not found` notification when a book quantity is one
- ensure no content response (204) is used to minimize browser console on pages or cases where no data are available
- ensure non-admin cannot see navigation link to admin page
- follow singleton in some cases for class exports and improve on variable naming
- account for errors in application, reduce redundancy and reuse codes, minimize unneccesary database calls, return appropriate error codes and make good use of destructuring throughout application
- follow single responsibility principle for functions

### How should this be manually tested?
1. Pull in latest changes from remote repository to local repository
2. Fire up terminal run `npm install`
3. execute `npm run start:dev` and then `npm run build:dev` to start up application
4. Visit url supplied by the application usually **http://localhost:3000**
5. login with `ekprogs@gmail.com` and `dayo` for admin access, while for normal user access simply register a new user on the application
6. After seeing shelf page, click on pagination component to ensure see pagination component not flickering
7. Navigate to admin page and delete all books, then go to different pages on the application and observe no huge footer margin
8. As for other fixes go through codebase to see

### Any background context you want to provide?
N/A

### What are the relevant pivotal tracker stories?
#153553440 Implement first code review feedback

### Screenshots (if appropriate)
N/A

### Questions:
N/A
